Meow and Purr emotes are no longer usable by all carbons

ValuedEmployee opened the pull request:
## About The Pull Request

Currently all carbons are able to meow and purr using the new Felinid tongue exclusive emotes. I'm changing how it works so it properly denies anyone without Felinid tongues from using the emotes. Mimes also will not be able to use the emote sounds so they stay silent. Lastly, there was a delay for the emotes on their first use where the player had to use them twice for it to work and I've fixed that, and I also removed some redundant text warnings for when you can't use the emote.

Edit: The Purr emote has been tweaked so that it isn't a visible emote and deaf players can't see it when it's used. Mimes can't purr anymore because it has no visible components so they shouldn't be able to.

## Why It's Good For The Game

Every carbon meowing and purring is really bad, and it was only approved to work for Felinids. This PR will rectify that. Multiple fixes and tweaks to how this emote should function are also huge positives.

:cl:
fix: Only players with Felinid tongues get to meow and purr for real this time, and other carbons cannot
fix: Mimes can no longer meow out loud and have their own mime message
fix: Stopped delay on meow and purr emotes where they had to be used twice to work the first time they're used
del: Removed redundant text warning for players that can't use the meow or purr emotes
tweak: Purring is no longer audible to deaf players
tweak: Players with vow of silence can't purr anymore
/:cl:

ValuedEmployee committed:
Adds meowing and purring back for felinids

Meow and purr emotes are added for felinid use only.

ValuedEmployee committed:
Warning and Requirement Change

Made it so you can meow and purr with a felinid tongue as opposed to being the felinid species. Also added a warning in chat if you can't meow or purr.

ValuedEmployee committed:
Fix

ValuedEmployee committed:
Merge branch 'master' into my-felinids-will-meow

ValuedEmployee committed:
Fix

ValuedEmployee committed:
Fixes fixes fixes

*Only felinids get to meow for real this time
*Mimes can't meow out loud
*Fixed delay on emote where it had to be used twice to work
*Removed redundant text warning

ValuedEmployee committed:
Merge branch 'master' into my-felinids-will-meow

Loafin34 commented:
The curse of the Cat will remain on those tainted by their bloodline. No son of man will be subject to it, and no sages of silence can break their vow so easily again.

Sadboysuss commented on code/modules/mob/living/carbon/emote.dm:
```diff
@@ -87,31 +87,29 @@
 	vary = TRUE
+	sound = SFX_CAT_PURR
+	message = "purrs."
+	message_mime = "purrs silently."
```
Not sure how you'd purr silently, maybe pantomime instead?

ValuedEmployee replied:
I'll try and think of something.

ValuedEmployee replied:
Maybe "imitates a tongue trill."?

Sadboysuss replied:
interesting problem cause if you do that then it'd make sense to be visible for mimes, so shrug

ValuedEmployee replied:
Maybe I could just remove the mime's capability.

ValuedEmployee replied:
Tried a fix for vow of silence users not being able to purr, looks like it's working on my test server.

Sadboysuss commented on code/modules/mob/living/carbon/emote.dm:
```diff
@@ -87,31 +87,29 @@
+	sound = SFX_CAT_PURR
+	message = "purrs."
+	message_mime = "purrs silently."
+	emote_type = EMOTE_VISIBLE | EMOTE_AUDIBLE
```
Probably not visible

ValuedEmployee replied:
This decides whether the emote has a visible and/or audible bitflag, so I want to use it in case. Other emotes like cry or snap have this.

ValuedEmployee replied:
That's a fair point, it probably doesn't even need that line then right?

Sadboysuss replied:
just leave `EMOTE_AUDIBLE` as `/datum/emote` has `var/emote_type = EMOTE_VISIBLE` by default

ValuedEmployee replied:
Alright, thanks!

grungussuss reviewed (commented):
Meowtainer

Ghommie commented:
Power tastes best when sweetened by catnip.

ValuedEmployee committed:
Purr changes

Purring will no longer be visible when deaf, and mimes get a more realistic imitation of purring.

ValuedEmployee committed:
Removes mime purring

Mimes can't purr anymore
